fulp_modules/features/antagonists/infiltrators/infil_items.dm

```diff
@@ -63,3 +63,83 @@
 	. = ..()
 	if (used)
 		. += "This one is all used up."
+
+
+/obj/item/infiltrator_radio
+	name = "Infiltrator Radio"
+	///List of objectives we have already checked
+	var/list/checked_objectives = list()
+	///Pool of rewards
+	var/reward_items = list(
+	/obj/item/card/emag,
+	/obj/item/card/emag/doorjack,
+	/obj/item/grenade/syndieminibomb,
+	/obj/item/grenade/c4/x4,
+	/obj/item/pen/edagger,
+	/obj/item/guardiancreator/tech/choose/traitor,
+	/obj/item/jammer,
+	/obj/item/reagent_containers/hypospray/medipen/stimulants,
+	/obj/item/storage/box/syndie_kit/imp_stealth,
+	)
+	desc = "How is this thing running without a battery?"
+	icon = 'fulp_modules/features/antagonists/infiltrators/icons/infils.dmi'
+	icon_state = "infiltrator_radio"
+	w_class = WEIGHT_CLASS_SMALL
+	///has the player claimed all his objectives' rewards?
+	var/objectives_complete = FALSE
+
+/obj/item/infiltrator_radio/proc/check_reward_status(mob/living/user)
```

Comment:
What if all infiltrator objectives send a signal to show it's completed an objective, that way we dont have to check every single ui_data for every objective in this guy's mind to see if any are done?